codegen/method.go

```diff
@@ -838,6 +840,74 @@ func (ms *MethodSpec) setDownstream(
 	return nil
 }
 
+func (ms *MethodSpec) isMethodprinted(methodName string) bool {
+	sort.Strings(ms.ConverterFunctionNames)
+	i := sort.SearchStrings(ms.ConverterFunctionNames, methodName)
+	if i < len(ms.ConverterFunctionNames) && ms.ConverterFunctionNames[i] == methodName {
+		return true
+	}
+	return false
+}
+
+func (ms *MethodSpec) genStructHelperFunctions(
+	typeConverter *TypeConverter,
+	downstreamMethod *MethodSpec,
+	requestType string,
+) error {
+	typeConverter.IsMethodCall = true
+	index := 0
+	for index < len(typeConverter.HelperFunctionStructs) {
+		helperStruct := typeConverter.HelperFunctionStructs[index]
+		index++
+		fromField := helperStruct.FromField
+		toField := helperStruct.ToField
+
+		methodName := "convertTo" + pascalCase(ms.Name) + pascalCase(fromField.Name) + requestType
+
+		// dedup method
+		if ms.isMethodprinted(methodName) {
+			continue
+		}
+		ms.ConverterFunctionNames = append(ms.ConverterFunctionNames, methodName)
+
+		// from field type
+		fromFieldType, err := typeConverter.getIdentifierName(fromField.Type)
+		if err != nil {
+			return err
+		}
+		fromFieldType = "*" + fromFieldType
```

Comment:
nit: maybe do a sanity check that `fromField.Type` is a struct before we prefix a `'*'`